pwiz_tools/Skyline/Model/Results/ScanProvider.cs

```diff
@@ -261,10 +295,14 @@ public void Dispose()
         {
             lock (this)
             {
-                if (_dataFile != null)
+                foreach (var key in _dataFileCentroidedMap.Keys.ToList())

```

Comment:
I am not sure I understand what is happening in this function.
Normally, you would call Dispose on all of the Value's in the Dictionary, and then clear the dictionary.
It looks instead like you are looking for one specific item in the dictionary, calling Dispose on that, and removing it from the Dictionary.
"Dispose" is supposed to be called on an object at the end of its life, so if "Dispose" is called on the ScanProvider, we would normally expect that this Dictionary would end up empty because the ScanProvider is no longer alive.